### PR TITLE
chore(local_proxy): reduce some startup logging

### PR DESCRIPTION
**Pull request by conradludgate**
Currently, local_proxy will write an error log if it doesn't find the config file. This is expected for startup, so it's just noise. It is an error if we do receive an explicit SIGHUP though.

I've also demoted the build info logs to be debug level. We don't need them in the compute image since we have other ways to determine what code is running.

Lastly, I've demoted SIGHUP signal handling from warn to info, since it's not really a warning event.

See https://github.com/neondatabase/cloud/issues/10880 for more details

### CODE REVIEW

File: proxy/src/bin/local_proxy.rs [hunk 1]
```diff
@@ -124,8 +125,8 @@ async fn main() -> anyhow::Result<()> {
 
     Metrics::install(Arc::new(ThreadPoolMetrics::new(0)));
 
-    info!("Version: {GIT_VERSION}");
-    info!("Build_tag: {BUILD_TAG}");
+    debug!("Version: {GIT_VERSION}");
```
Comment (cloneable) [inline proxy/src/bin/local_proxy.rs:128]: I actually like having these in the logs, but could be on one like.

**Review by cloneable — APPROVED** [review]

File: proxy/src/bin/local_proxy.rs (hunk 1 above)
Comment (conradludgate) [reply proxy/src/bin/local_proxy.rs:128]: This is local proxy only ^^

Comment (conradludgate) [reply proxy/src/bin/local_proxy.rs:128]: More specifically, the current local_proxy logs show:
```
INFO Version: git:unknown
INFO Build_tag: build_tag-env:6504
```

Neither of these are useful 😅

Comment (conradludgate) [reply proxy/src/bin/local_proxy.rs:128]: Should we desire to fix this, I think we should move it to compute_ctl instead.

**Pull request merged**